framework/src/modules/chain/submodules/blocks/process.js

```diff
@@ -240,108 +242,6 @@ __private.receiveForkFive = function(block, lastBlock, cb) {
- * @param {function} cb - Callback function
- * @returns {function} cb - Callback function from params (through setImmediate)
- * @returns {Object} cb.err - Error if occurred
- * @returns {Object} cb.res - Result object
- */
-Process.prototype.getCommonBlock = function(peer, height, cb) {
-	let comparisonFailed = false;
-
-	return async.waterfall(
-		[
-			function(waterCb) {
-				// Get IDs sequence (comma separated list)
-				modules.blocks.utils.getIdSequence(height, (err, res) =>
-					setImmediate(waterCb, err, res)
-				);
-			},
-			function(res, waterCb) {
-				const ids = res.ids;
-				// Perform request to supplied remote peer
-				peer = library.logic.peers.create(peer);
-				peer.rpc.blocksCommon({ ids }, (err, blocksCommonRes) => {
-					if (err) {
-						modules.peers.remove(peer);
-						return setImmediate(waterCb, err);
-					}
-
-					if (!blocksCommonRes.common) {
-						// FIXME: Need better checking here, is base on 'common' property enough?
-						comparisonFailed = true;
-						return setImmediate(
-							waterCb,
-							`Chain comparison failed with peer: ${
-								peer.string
-							} using ids: ${ids}`
-						);
-					}
-
-					return setImmediate(waterCb, null, blocksCommonRes.common);
-				});
-			},
-			function(common, waterCb) {
-				// Check if we received genesis block - before response validation, as genesis block have previousBlock = null
-				if (common && common.height === 1) {
-					comparisonFailed = true;
-					return setImmediate(
-						waterCb,
-						'Comparison failed - received genesis as common block'
-					);
-				}
-				// Validate remote peer response via schema
-				return library.schema.validate(common, definitions.CommonBlock, err => {
-					if (err) {
-						return setImmediate(waterCb, err[0].message);
-					}
-					return setImmediate(waterCb, null, common);
-				});
-			},
-			function(common, waterCb) {
-				// Check that block with ID, previousBlock and height exists in database
-				library.storage.entities.Block.isPersisted({
-					id: common.id,
-					previousBlockId: common.previousBlock,
-					height: common.height,
-				})
-					.then(isPersisted => {
-						if (isPersisted) {
-							// Block exists - it's common between our node and remote peer
-							return setImmediate(waterCb, null, common);
-						}
-
-						// Block doesn't exists - comparison failed
-						comparisonFailed = true;
-						return setImmediate(
-							waterCb,
-							`Chain comparison failed with peer: ${
-								peer.string
-							} using block: ${JSON.stringify(common)}`
-						);
-					})
-					.catch(err => {
-						// SQL error occurred
-						library.logger.error(err.stack);
-						return setImmediate(waterCb, 'Blocks#getCommonBlock error');
-					});
-			},
-		],
-		(err, res) => {
-			// If comparison failed and current consensus is low - perform chain recovery
-			if (comparisonFailed && modules.transport.poorConsensus()) {
-				return modules.blocks.chain.recoverChain(cb);
```

Comment:
this logic needs to stay

Reply:
#3281

Reply:
Ah yep, we need the `recoverChain(...)` feature but it should be possible to use it as part of the `loadBlocksFromNetwork` logic instead - It would be nice to remove the `commonBlocks` logic.